connectors/sources/tests/test_servicenow.py

```diff
@@ -0,0 +1,343 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the ServiceNow source class methods"""
+from unittest import mock
+
+import pytest
+from aiohttp.client_exceptions import ServerDisconnectedError
+
+from connectors.source import ConfigurableFieldValueError, DataSourceConfiguration
+from connectors.sources.servicenow import ServiceNowClient, ServiceNowDataSource
+from connectors.sources.tests.support import create_source
+from connectors.tests.commons import AsyncIterator
+
+
+class MockResponse:
```

Comment:
Could use an `AsyncMock()` instead of a custom class?

Reply:
Custom class still be needed after extending `AsyncMock`. So created custom class with `aenter` & `aexit`